### ISSUE TITLE
App Service sync fails

### ISSUE THREAD

**Issue by wly-saa** [OfficeDev/microsoft-teams-stickers-app#43]
Hello, I've been having some problems with the app. in the teams client i get an error stating the app is unreachable. after some troubleshooting I figured i would run a sync in case there was a change that wasn't applied but the sync is failing.

In one of the other posts, you recommended double checking the botID in the manifest.json. I double checked and its correct, fiddler showed a call out to the app that returned a 502 error :
{"errorCode":1008,"message":"<BotError>Bot returned unsuccessful status code Forbidden"}

the request header has the correct applicationID in it so i'm assuming its correct:
Host: amer.ng.msg.teams.microsoft.com
POST /v1/agents/28:_clientApplicationID_/invoke HTTP/1.1

attached is the deployment failure log from azure and a couple of screenshots.

Any help is appreciated thank you!

[failedLog.txt](https://github.com/OfficeDev/microsoft-teams-stickers-app/files/4704823/failedLog.txt)

![teamsError](https://user-images.githubusercontent.com/65689134/83314349-d0677f00-a1ce-11ea-93e9-e4b8897d052b.PNG)
![azurePortal](https://user-images.githubusercontent.com/65689134/83314350-d1001580-a1ce-11ea-803a-74b90b3393be.PNG)

**Comment by rashmi-dixit** [comment]
Was the deployment successful? If so, there would be another row in the deployment center with status as "Success(Active)"

**Comment by wly-saa** [comment]
you're right, I think it failed and I redeployed it without deleting the resources. It would seem when ever I attempt to deploy the app it fails at the step where it creates Microsoft.Web/sites/sourcecontrols
shows a Conflict (checked to make sure the base name was free) for the client app.
{
    "status": "Failed",
    "error": {
        "code": "ResourceDeploymentFailure",
        "message": "The resource operation completed with terminal provisioning state 'Failed'."
    }
}

**Comment by LindaSchaar** [comment]
We have the same issues, after step 2 deploy to azure we receive deployment failed. Failed: code: ResourceDeploymentFailure

**Comment by wly-saa** [comment]
i was able to make it work by changing the StickersTemplate Microsoft.NET.Sdk.Functions from 1.0.24 to 1.0.26

**Comment by LindaSchaar** [comment]
Nice! how did you change it?

**Comment by LindaSchaar** [comment]
Ok because after 2 days successfully deploy at Azure when I try to contact de app in a team I see this:
![image](https://user-images.githubusercontent.com/49586227/84468780-ed488b80-ac7f-11ea-831e-8cc68c67aec9.png)
Unable to reach app, please try again. So you say start over again? registration app and deploy to azure?

**Comment by rashmi-dixit** [comment]
Thank you for reporting this, we have added this to the list of issues to fix. If you do fix it before we get to it, please help the community by raising a pull request with your fix!

**Comment by kunalkankariya** [comment]
@rashmi-dixit ,

This is the error from the browser network trace. The bot framework's "api/messages/" endpoint is not reachable as well.

{"errorCode":1008,"message":"<BotError>Bot returned unsuccessful status code NotFound"}

**Comment by rashmi-dixit** [comment]
Can you please confirm if the deployment was successful?

**Comment by LindaSchaar** [comment]
I will try the registration and deployment at a later moment. Than I will chat you if the stickers! app install was succesfull in teams

**Comment by rashmi-dixit** [comment]
If it is already deployed, please check in the app service's deployment center if status is "Success(Active)"

**Comment by kunalkankariya** [comment]
Deployment Failed

**Comment by LindaSchaar** [comment]
@rashmi-dixit yes azure status succeeded. In teams via app studio, manifest, install app, after installing i receive unable to reach app. So why teams would not recognized the app, when azure says succeeded?

**Comment by kunalkankariya** [comment]
I concur with @wly-saa . Updating the package version of the  Microsoft.NET.Sdk.Function to 1.0.26 does fix this. Thanks @wly-saa . May I ask how did you get to this? The error does specify the issue with extension metadata generator but how did you figure updating the version is a solution? Thanks :)

**Comment by wly-saa** [comment]
Honestly some google-fu on the logs from the failed deployment of the app service. Saw a suggestion on a forum with the same error and i just gave it a shot and it worked.
@LindaSchaar did you check each of the services? I found that the App Service for the actual messaging app (not the config app) failed. you'll see it under deployment center.

While testing I found that the initial deployment failed, and I would hit redeploy and it would succeed but the app service deployment still showed failed in the Deployment Center blade.

**Comment by kunalkankariya** [comment]
@LindaSchaar or you can delete both the app services and redeploy using this git url https://github.com/kunalkankariya/microsoft-teams-stickers-app-master

**Comment by rashmi-dixit** [comment]
@wly-saa, @kunalkankariya - can you please raise a pull request since it seems if you have a fix for the issue?

**Comment by kunalkankariya** [comment]
@wly-saa  - Aren't you able to push your commit?

**Comment by yeahsmaggy** [comment]
@kunalkankariya does your repo link above https://github.com/kunalkankariya/microsoft-teams-stickers-app-master.git fix the "Unable to reach app. Please try again" issue in Teams? @LindaSchaar  did you get it working?

**Comment by kunalkankariya** [comment]
@yeahsmaggy  - I was able to fix issues and get the app up and running. It works in my test tenant. My repo also add animation feature to the stickers.

**Comment by yeahsmaggy** [comment]
@kunalkankariya  so do I need to start from the beginning with the deployment steps but use your repo? How come the OfficeDev team haven't fixed the issue in the official repo?

**Comment by kunalkankariya** [comment]
@yeahsmaggy - What you need to do is follow all the deployment steps from here and when that's done, go to the Azure Portal and find the config app service. In the deployment center section, change the repo to my repo and you should be good.

I am not sure why the OfficeDev team won't approve the changes proposed.

**Comment by yeahsmaggy** [comment]
@kunalkankariya  I already followed the deployments steps. Then I went to the config app service in the deployment center section and I disconnected the existing one. Then I added a new one with your repo. Any other steps as so far it has no changed the issue I have in Teams "Unable to reach app. Please try again" - perhaps I have some other issue ?

**Comment by yeahsmaggy** [comment]
@kunalkankariya  it is working for me.

**Comment by kunalkankariya** [comment]
@yeahsmaggy - Glad to know that Andy! Did you face any other issues? I assume it works with my repo. You should also be able to use animated stickers (gifs). Thanks.
